utils/conversions.ts

```diff
@@ -115,33 +127,61 @@ export const toApiUserBonus = (domainObj: UserBonus): BonusActivationItem => {
   };
 };
 
-function calculateMaxBonusAmount(
-  numberOfFamilyMembers: number
+/**
+ * Returns the maximum bonus amount in Euros from the total number of family
+ * members.
+ */
+function calculateMaxBonusAmountFromTotFamilyMembers(
+  numberOfFamilyMembers: FamilyMemberCount
 ): MaxBonusAmount {
-  return (numberOfFamilyMembers > 2
-    ? 500
-    : numberOfFamilyMembers === 2
-    ? 250
-    : numberOfFamilyMembers === 1
-    ? 150
-    : 0) as MaxBonusAmount;
+  if (numberOfFamilyMembers > 2) {
+    return THREE_OR_MORE_FAMILY_MEMBERS_AMOUNT;
+  }
+  if (numberOfFamilyMembers === 2) {
+    return TWO_FAMILY_MEMBERS_AMOUNT;
+  }
+  if (numberOfFamilyMembers === 1) {
+    return ONE_FAMILY_MEMBER_AMOUNT;
+  }
+  throw new Error(
+    `FATAL: number of family members is 0 or negative [${numberOfFamilyMembers}]`
+  );
 }
 
+/**
+ * Calculate
+ */
 function calculateMaxBonusTaxBenefit(
   maxBonusAmount: MaxBonusAmount
 ): MaxBonusTaxBenefit {
-  return (maxBonusAmount / 5) as MaxBonusTaxBenefit;
+  return ((TAX_BENEFIT_PERCENT * maxBonusAmount) / 100.0) as MaxBonusTaxBenefit;
 }
 
 export const toEligibilityCheckFromDSU = (
   data: ConsultazioneSogliaIndicatoreResponse,
   fiscalCode: FiscalCode,
   validBefore: Timestamp
 ): ApiEligibilityCheck => {
-  const bonusValue = calculateMaxBonusAmount(
+  const validatedNumberOfFamilyMembers = FamilyMemberCount.decode(
     data.DatiIndicatore?.Componenti ? data.DatiIndicatore.Componenti.length : 0
   );
 
+  if (isLeft(validatedNumberOfFamilyMembers)) {
+    return EligibilityCheckFailure.encode({
+      error: ErrorEnum.INTERNAL_ERROR,
+      error_description: `Family members count is out of range [${readableReport(
+        validatedNumberOfFamilyMembers.value
+      )}]`,
+      id: (fiscalCode as unknown) as NonEmptyString,
+      status: ErrorStatusEnum.FAILURE
+    });
+  }
+  const numberOfFamilyMembers = validatedNumberOfFamilyMembers.value;
+
+  const bonusValue = calculateMaxBonusAmountFromTotFamilyMembers(
+    numberOfFamilyMembers
+  );
+
   const familyMembers: FamilyMembers = data.DatiIndicatore?.Componenti
     ? rights(
```

Comment:
@gunzip I'm not sure it makes sense to just ignore invalid values, it's probably better to make the request fail, at least we can catch the issue, or else the user will see some different info in respect to what he sees on the INPS site and we will not have any evidence of the problem in the logs

Reply:
that's right I've created https://www.pivotaltracker.com/story/show/173309312